MAUI/DataGrid/Column-sizing.md

```diff
@@ -0,0 +1,186 @@
+
+{% tabs %}
+{% highlight xaml %}
+<syncfusion:SfDataGrid  x:Name="dataGrid"
+                        ColumnWidthMode="Fill"
+                        ItemsSource="{Binding Orders}" />
+{% endhighlight %}
+{% endtabs %}
+
+![Column Sizing in .NET MAUI DataGrid](Images\column-sizing\columnWidthModeFill.png)
+
+
+## Apply ColumnWidthMode for a particular column
+
+To apply column sizing to individual column, use the `DataGridColumns.ColumnWidthMode` property. The `GridColumn.ColumnWidthMode` property is also a type of the `ColumnWidthMode`. If the `GridColumn.ColumnWidthMode` is not explicitly set to a value, then it takes the value of the `SfDataGrid.ColumnWidthMode`and applies width to the columns accordingly.
+
+To apply `ColumnWidthMode` for a particular column, follow the code example:
+
+{% tabs %}
+{% highlight xaml %}
+<sfgrid:SfDataGrid x:Name="dataGrid"
+                   AutoGenerateColumns="True"
+                   ColumnWidthMode="None">
+
+   <syncfusion:DataGridTextColumn MappingName="CustomerID" ColumnWidthMode="Auto" HeaderText="Customer" ></syncfusion:DataGridTextColumn>
+{% endhighlight %}
+{% highlight c# %}
+DataGridTextColumn textColumn = new DataGridTextColumn();
+textColumn.MappingName = "CustomerID";
+textColumn.HeaderText = "Customer";
+textColumn.ColumnWithMode = ColumnWithMode.Auto;  
+{% endhighlight %}
+{% endtabs %}
+
+![Column Sizing in .NET MAUI DataGrid](Images\column-sizing\particularColumn.png)
+
+## Refreshing ColumnSizer at runtime
+
+To refresh the column sizing for `SfDataGrid.Columns` at runtime, use the `SfDataGrid.ColumnSizer.Refresh` method.
+
+Consider that `ColumnWidthMode.Auto` is applied to the SfDataGrid. If the underlying values are changed at run time, refresh the column sizer as follows:
+
+{% tabs %}
+{% highlight xaml %}    
+<StackLayout HorizontalOptions="Center" 
+             Orientation="Vertical">
+    <Button x:Name="button"
+            Text="Refresh ColumnSizer"
+            HeightRequest="100"
+            HorizontalOptions="Center"
+            Clicked="ColumnSizerChanged"/>
+<sfgrid:SfDataGrid x:Name="dataGrid"
+                   ColumnWidthMode="Auto"
+                   ItemsSource="{Binding OrdersInfo}">
+</sfgrid:SfDataGrid>
+</StackLayout>
+{% endhighlight %}
+{% highlight c# %}  
+private void ColumnSizerChanged(object sender, EventArgs e)
+{
+    //Refreshes the column sizer of the SfDataGrid
+    dataGrid.GridColumnSizer.Refresh(true);
+}    
+{% endhighlight %}
+{% endtabs %} 
+ 
+
+## Customize auto width calculation for a column
+
+For cases, where a column might require more width than the applied auto width or if you want to apply your own custom logic to calculate the auto width of a column, return a desired width in the `OnComputeCellWidth()`override of the custom written column-sizer class derived from `DataGridColumnSizer` and assign it to the `SfDataGrid.ColumnSizer` property.
```

Comment:
For cases where a column might require more width than the applied auto width or if you want to use your custom logic to calculate the auto width of a column, return a desired width in the `OnComputeCellWidth()`override of the custom-written column-sizer class derived from `DataGridColumnSizer` and assign it to the `SfDataGrid.ColumnSizer` property.